pkg/eventemitter/eventemitter.go

```diff
@@ -0,0 +1,247 @@
+/*
+Copyright 2022 The KEDA Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package eventemitter
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/tools/record"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+
+	kedav1alpha1 "github.com/kedacore/keda/v2/apis/keda/v1alpha1"
+)
+
+var log = logf.Log.WithName("event_emitter")
+var ch chan EventData
```

Comment:
I think that this channel should be scoped to EventEmiter instead of scoped to the package. This allows us to init multiple EventEmiters in the future is it's required

Reply:
I am adding a log Initializer for EventEmiters. And as eventemitter is a global handler, I think it is OK to keep it scoped to the package?

Reply:
When these vars are scoped to the package, they should have more descriptive name. So that in half a year we still understand what they're for.